src/sidebar/services/groups.js

```diff
@@ -348,25 +323,115 @@ export default function groups(
       directLinkedGroupId
     );
 
+    const groupToFocus =
+      directLinkedAnnotationGroupId || directLinkedGroupId || null;
+    addGroupsToStore(groups, groupToFocus);
+
+    return groups;
+  }
+
+  /**
+   * Load the specific groups configured by the annotation service.
+   *
+   * @param {string[]} groupIds
+   */
+  async function loadServiceSpecifiedGroups(groupIds) {
+    // Fetch the groups that the user is a member of in one request and then
+    // fetch any other groups not returned in that request directly.
+    //
+    // This reduces the number of requests to the backend on the assumption
+    // that most or all of the group IDs that the service configures the client
+    // to show are groups that the user is a member of.
+    const userGroups = await api.profile.groups.read({ expand: expandParam });
+
+    let error;
+    const tryFetchGroup = async id => {
+      try {
+        return await fetchGroup(id);
+      } catch (e) {
+        error = e;
+        return null;
+      }
+    };
+
+    const getGroup = id =>
+      userGroups.find(g => g.id === id || g.groupid === id) ||
```

Comment:
Can you help me understand why you need the `g.groupid` check in this? Are there situations in which `groupid` is set but not `id`?

Reply:
`id` may be an ID or a `groupid` here because the service config can specify either group IDs or group `groupid`s. Therefore when looking up the ID in the results of `/api/profile/groups` we need to check both properties.

Does this clarify things?

Reply:
Yes, this makes sense. Could you please update the JSdoc string for `@param {string[]} groupIds` to reflect this information? I could see it confusing folks later on.